docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+
+The policy file must be generated and maintained by the user. Additional security means need to be used to ensure that the policy file will not be modified by malicious entities during storage and transmission. If users have high requirements for security level, they even need to use additional means to ensure the confidentiality of policy files.
+
+If image-rs finds that there is no policy.json in the /etc/containers/ directory or If the policy.json file is empty, call the GetResource grpc service provided by the attestation-agent. After remote attestation, obtain it from the user controlled relying KBS through the encrypted channel.
+
+The format of the policy.json file follows the format and syntax defined in the [container/image policy.json](https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md) document.
+
+In the V1 design of the confidential containers, a safe and reasonable policy.json file may look like the following example: 
+
+```json
+{
+    "default": [{"type": "reject"}], 
+    "transports": {
+        "docker": {
+            "docker.io/my_private_registry": [
+                {
+                    "type": "signedBy",
+                    "keyType": "GPGKeys",
+                    "keyData": "<public Key>",
+                }
+            ],
+            "registry.access.redhat.com": [
+                {
+                    "type": "signedBy",
+                    "keyType": "GPGKeys",
+                    "keyPath": "/etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release",
+                }
+            ],
+        },
+        "dir": {
+            "": [{"type": "insecureAcceptAnything"}]
+        }
+    }
+}
+```
+
+Before pulling the container image layers, match the policy in the following order according to the location/registry of the container image and execute the corresponding policies:
+
+- Match single specific scopes under each member in transport, for example:
+  - For the image from the container registry that implements the Docker Registry HTTP API V2 protocol, match each scope under the "docker" member;
+  - For the image from local file system, match each scope under the "dir" member;
+- Match the default scope under the member in the transport. for example, the "" scope of the transport "dir".
+- Match the global default scope, such as the "default" field in the first line of the above example. We suggest that this policy must be configured to reject for security.
+
+After the matching is successful, the policy set under the corresponding scope is executed. In the design of confidential containers V1, only three policy types are currently supported now: "reject", "insecureAcceptAnything" and "signedBy". The former two is as the name implies, and the "signedBy" policy requires that there must exist at least one signature of this image can be verified by the public key ring provided by `keyPath` or `keyData` filed. In CCv1, he format of "signedBy" policy is defined as follows:
```

Comment:
How is an image resolved if multiple registries are specified and they all contain the same image? Will the local image be used in preference to the registry version?

How does "generic matching" affect this? `transports` is an object rather than an array, so are the entries searched in random order and the first matching image used?